karrot/activities/serializers.py

```diff
@@ -391,25 +416,31 @@ def get_attendee(self, activity):
 class ActivityJoinSerializer(serializers.ModelSerializer):
     class Meta:
         model = ActivityModel
-        fields = []
+        fields = [
+            'role',
+        ]
+
+    role = CharField(write_only=True)
 
     def update(self, activity, validated_data):
         user = self.context['request'].user
-        activity.add_participant(user)
+        place = activity.place
+        group = place.group
+        activity.add_participant(user, validated_data.get('role', None))
```

Comment:
I think need a new permission check to make sure the user actually has that role, either here, or in a permission class.